src/components/restaurant/Restaurant.js

```diff
@@ -0,0 +1,100 @@
+// Libraries
+import React from "react";
+
+// lodash
+import _isEmpty from "lodash/isEmpty";
+
+// components
+import SubHeader from "../subHeader";
+import FoodDescription from "../foodDescription";
+import Loader from "../../commonComponents/loader";
+import ErrorHandler from "../../commonComponents/errorHandler";
+
+// constants
+import {
+    INITIAL_STATE,
+    DEFAULT_RESTAURANT_ID,
+    GENERAL_ERROR_MESSAGE,
+} from "./constants/restaurant.general";
+
+// helpers
+import fetchRestaurantData from "../../api/fetchRestaurantData";
+
+// readers
+import restaurantReader from "./readers/restaurantReader";
+
+// css
+import "./restaurant.css";
+
+class Restaurant extends React.Component {
+    state = INITIAL_STATE;
+
+    componentDidMount() {
+        this.getRestaurantData();
+    }
+
+    getRestaurantData() {
+        fetchRestaurantData(DEFAULT_RESTAURANT_ID)
+            .then(this.setRestaurantData)
+            .catch(this.setErrorMessage)
+            .finally(this.setLoaded);
+    }
+
+    setRestaurantData = (restaurantData) => {
+        const restaurantDetails =
+            restaurantReader.restaurantDetails(restaurantData);
+        const foodList = restaurantReader.foodList(restaurantData);
+        this.setState({
+            restaurantDetails,
+            foodList,
+        });
+    };
+
+    setErrorMessage = (err) => {
+        const { message = GENERAL_ERROR_MESSAGE } = err;
+
+        this.setState({ error: message });
+    };
+
+    setLoaded = () => {
+        this.setState({ isLoading: false });
+    };
+
+    renderSubheaderAndFoodDescription() {
+        const { restaurantDetails } = this.state;
+
+        if (_isEmpty(restaurantDetails)) {
```

Comment:
const isRestaurantDetailsEmpty = _isEmpty(restaurantDetails);
write conditions like that benefits

1. it's more scalable so tomorrow we can add multiple conditions and it still readable instead of inferring
2. If condition grows it can be moved out into a function
3. in general it's more readable